model/registry/hardware.yaml

```diff
@@ -0,0 +1,113 @@
+groups:
+  - id: registry.hardware
+    prefix: hw
+    type: attribute_group
+    brief: >
+      Attributes for hardware.
+    attributes:
+      - id: id
+        type: string
+        stability: experimental
+        brief: >
+          An identifier for the hardware component, unique within the monitored host
+        examples: ['win32battery_battery_testsysa33_1']
+      - id: name
+        type: string
+        stability: experimental
+        brief: >
+          An easily-recognizable name for the hardware component
+        examples: ['eth0']
+      - id: parent
```

Comment:
```suggestion
      - id: parent.id
```

Reply:
This will introduce a breaking change but probably we can't go without one as `parent` is very common

Reply:
Personal opinion: Let's not put namespaces everywhere. Will there ever be `parent.name`, or `parent.parent`?

`parent` is indeed very generic. If it belongs to the hardware semantic conventions only, it could be later changed to `hw.parent`. 

However, I think this attribute could be beneficial to other semconv, like process, system disks and filesystems, databases, etc. Maybe it could be specified as a general attribute, applicable everywhere.

This will be discussed in a separate issue.

Reply:
@bertysentry there is a existing `process.parent` fields from ECS. But I suspect parent could be used also somewhere else. So I am also for introducing a separate namespace for it. But I don't want to do this in this PR. I want this PR to be as plain as possible - only move things out to the yaml files and the registry without functional changes

Reply:
Understood, thank you @trisch-me!